### ISSUE TITLE
HDF5 filters not used

### ISSUE THREAD

**Issue by vuillaut** [cta-observatory/cta-lstchain#352]
I have a weird bug with the filters in the HDF5 writer of DL1:
When launching the script `lstchain_mc_r0_to_dl1`

```
    print(filters)

    writer = HDF5TableWriter(
        filename=output_filename,
        group_name='dl1/event',
        mode='a',
        filters=filters,
        add_prefix=True,
        # overwrite=True,
    )

    with writer:
        print("USING FILTERS: ", writer._h5file.filters)
```
it prints:
```
Filters(complevel=5, complib='blosc:zstd', shuffle=True, bitshuffle=False, fletcher32=True, least_significant_digit=None)
USING FILTERS:  Filters(complevel=0, shuffle=False, bitshuffle=False, fletcher32=False, least_significant_digit=None)
```

However, I cannot reproduce the bug with a simple script only making a writer.

I checked and it's not a problem in my install since the environment used in the cluster gives the same.

This means that all the DL1 files have no compression enabled.

I am trying to find the root of the problem but any help is welcome.

**Comment by MaxNoe** [comment]
Could you enable debug logging for the hdf writer? It logs the kwargs.

```
    def open(self, filename, **kwargs):

        self.log.debug("kwargs for tables.open_file: %s", kwargs)
        self._h5file = tables.open_file(filename, **kwargs)

```

**Comment by MaxNoe** [comment]
@vuillaut I think the problem is that that the call to `write_array_info` is missing the filters, so the file get's created without them.

**Comment by MaxNoe** [comment]
Are you on the current master? It should already be fixed there

**Comment by MaxNoe** [comment]
In #348, I moved the order of the writes to not rely on an additional read of the first event, and this should also have fixed this issue

**Comment by vuillaut** [comment]
Hi @MaxNoe, thanks for the feedback.
To your questions:
- I am using master
- it does not seem to come from `write_array_info` as commenting it does not change the issue. Plus I fail to understand how a line coming after the writer instantiation would affect it. It can try commenting a bunch of things to see what can affect it however.

**Comment by MaxNoe** [comment]
> it does not seem to come from write_array_info as commenting it does not change the issue. Plus I fail to understand how a line coming after the writer instantiation would affect it.

Before the changes in #341, the write_array_info came before the hdfwriter was instantiated, this is exactly why I asked if you were using the master.

**Comment by MaxNoe** [comment]
Does the output file already exist?

**Comment by vuillaut** [comment]
> Does the output file already exist?

No, the script would fail from start otherwise:
```
    if os.path.exists(output_filename):
        raise IOError(output_filename + ' exists, exiting.')
```

**Comment by MaxNoe** [comment]
But the reason is somewhat related, the file already exists before the hdfwriter is created.

So it's not the write_array info but something else probably.

**Comment by MaxNoe** [comment]
It's the write_metadata

**Comment by vuillaut** [comment]
> It's the write_metadata

I just arrived at the same conclusion, well spotted!
I'll work on a patch.
Thank you for the feedback.

**Comment by vuillaut** [comment]
Although, it raises a question you might be able to answer.
HDF5 is able to compress per dataset, right?
So this means that the `HDF5Writer` is not able to allow a different compression per dataset?

**Comment by MaxNoe** [comment]
Yes, hdf supports compression per dataset, I don't know if the HDFWriter supports that.

**Comment by MaxNoe** [comment]
Fix for ctaipe in https://github.com/cta-observatory/ctapipe/pull/1285 allows to set compression filters per table by passing filters to `HDFTableWriter.write`

**Comment by rlopezcoto** [comment]
Shall we keep this opened for the update to the next ctapipe release or can we just close it via #353 ?

**Comment by MaxNoe** [comment]
This should be fixed by #353

**Comment by MaxNoe** [comment]
Let's start a general issue about porting to ctapipe 0.8 with a long list of ToDos
